### PR TITLE
VACMS-8550: hide old service location field_facility_service_hours

### LINKED ISSUE #8550: The old field_facility_service_hours field should be hidden from paragraph edit/displays and views
## Acceptance Criteria

- [ ] Hide the old Service Location field_facility_service_hours (table field) on edit/view. 
- [ ] The non-clinical services view needs to be updated to use the new field /admin/structure/views/view/non_clinical_services

## Implementation steps

Once the script from #6931 has been run on production and the FE team has built the new display to use the new office hours field: [Update VAMC Facility Health Service front end to utilize new office hours field](https://github.com/department-of-veterans-affairs/va.gov-team/issues/39436)

- [ ] Deploy this update (hides old field)
- [ ] Re-run script from #6931 

Do we need a separate ticket to remove the old field or should this ticket simply remove it instead of hiding it @swirtSJW

### PR DESCRIPTION
**Pull request by JayDarnell**
## Description

Closes #8550 

## Testing done

Visual Testing

## Screenshots

Screen Shot 2022-03-30 at 11 31 04 AM

Screen Shot 2022-03-30 at 11 30 56 AM

Screen Shot 2022-03-30 at 11 30 49 AM

Screen Shot 2022-03-30 at 11 31 45 AM

## QA steps

- [ ] View /eastern-colorado-health-care/register-for-care-0 (service location should show time)
- [ ] View /eastern-colorado-health-care/medical-records-office-0 (service location should show time)
- [ ] View /eastern-colorado-health-care/billing-and-insurance-0 (service location should show time)
- [ ] Edit a non-clinical service node (example: /node/42057/edit) 
  - [ ] the old table field for hours should no longer be visible
  - [ ] the new office hours field should be visible
  - [ ] the **Use the facility's hours** field should toggle the new office hours field visibility

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide CMS Team`
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS Experience`
  - [ ] `⭐️ Offices`
  - [x] `⭐️ Product Support`

### CODE REVIEW

**Comment by JayDarnell** [comment]
Fixed broken behat test

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -1171,17 +1171,6 @@ function va_gov_backend_field_widget_paragraphs_form_alter(&$element, FormStateI
       ];
     }
   }
-
-  // If we have a field ref'ing service location paragraphs, fire function.
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:1175]: I'm not clear on why this is all going away.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 2]
```diff
@@ -1291,59 +1280,6 @@ function _va_gov_backend_modify_contact_information_states(array &$element, Form
   }
 }
 
-/**
- * Toggles service_location fields depending on parent selector.
- *
- * Adds Validation constraint error id match to html.
- *
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:1298]: I am not clear on why this is all going away.   It should be altered for the new field, but it seems like removing all this breaks some unrelated stuff.

**Comment by swirtSJW** [comment]
This is blocked by https://github.com/department-of-veterans-affairs/va.gov-team/issues/39436 and can not be merged until that has been resolved.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module (hunk 2 above)
Comment (JayDarnell) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:1298]: The section above appeared to call _va_gov_backend_modify_paragraph_interactions for four paragraph fields but within the function call itself (below)  only performed work on a single paragraph type: service_location.

The work performed for this paragraph type was to set some attributes and the visibility for the field we are now hiding by default (which will eventually be removed): field_facility_service_hours. With this toggle logic removed the new field_office_hours field still respects the selection for the field_use_facility_address field so that logic must be located elsewhere.

Comment (swirtSJW) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:1298]: Hmm I would have expected the logic to be altered to control the new field's visibility based on the state of the selected option for use of hours.

Comment (JayDarnell) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:1298]: Yes I find this very confusing. My plan was initially to alter this for the new field but I noticed the new field was already honouring the toggle despite not being explicitly named in the $toggle_fields array below. I cannot find the controlling code for this.

**Comment by swirtSJW** [comment]
This has to follow https://github.com/department-of-veterans-affairs/content-build/pull/1080  on the same day.

**Comment by swirtSJW** [comment]
Tested on facility health services.
image

image

**Comment by swirtSJW** [comment]
This is unblocked and after the conflict resolved can be queued up for merge tomorrow.
 We will also need to re-run the migration script right after this goes out.

**Review by swirtSJW — APPROVED** [review]
image

image


All good.
